### PR TITLE
Translation fallbacks

### PR DESCRIPTION
**Pull request by magnearun**
Handle translation fallbacks in translation service

## What

Handle translation fallbacks in translation service, before this change the fallback support was handled in the Contentful extension. Since fallbacks are not set for locales in Contentful for various reasons (it was causing bugs on the website), I declared the used locales and fallbacks in code in the translation service, but using the same format as Contentful so this can be replaced if we set the fallbacks in contentful in the future.

## Why

If requested locale is English but the is not translation found, the icelandic string will be returned for that key.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against master before asking for a review

### CODE REVIEW

File: libs/api/domains/translations/src/lib/translations.service.ts [hunk 1]
```diff
@@ -23,19 +30,25 @@ export class TranslationsService {
     namespaces?: string[],
     lang?: string,
   ): Promise<TranslationsDict | null> => {
+    const locale = locales.find((l) => l.code === localeMap[lang])
+
     const result = await this.contentfulRepository
-      .getLocalizedEntries<any>(lang, {
+      .getLocalizedEntries<any>('*', {
```
Comment (arnifreyrs) [inline libs/api/domains/translations/src/lib/translations.service.ts:36]: so this fetches everything?

File: libs/api/domains/translations/src/lib/translations.service.ts [hunk 2]
```diff
@@ -23,19 +30,25 @@ export class TranslationsService {
...
       })
       .catch(errorHandler('getNamespace'))
 
-    const withFallbacks = result?.items?.map(({ fields }) =>
-      mergeWith({}, fields.fallback, fields.strings, (o, s) =>
-        isEmpty(s) ? o : s,
-      ),
-    )
+    const withFallbacks = result?.items?.map(({ fields }) => {
+      console.log('fields', fields)
```
Comment (arnifreyrs) [inline libs/api/domains/translations/src/lib/translations.service.ts:44]: away with you

File: libs/application/templates/reference-template/src/forms/messages.ts [hunk 3]
```diff
@@ -13,7 +13,8 @@ export const m = defineMessages({
   },
   introIntroduction: {
     id: 'example.application:intro.introduction',
-    defaultMessage: 'Halló {name}',
+    defaultMessage:
```
Comment (arnifreyrs) [inline libs/application/templates/reference-template/src/forms/messages.ts:16]: this should not be part of the PR

File: libs/application/ui-fields/src/lib/IntroductionFormField.tsx [hunk 4]
```diff
@@ -21,7 +22,9 @@ const IntroductionFormField: FC<{
           {formatText(field.name, application, formatMessage)}
         </Text>
       )}
-      <Text>{formatText(field.introduction, application, formatMessage)}</Text>
+      <Markdown>
```
Comment (arnifreyrs) [inline libs/application/ui-fields/src/lib/IntroductionFormField.tsx:25]: nor this

**Review by arnifreyrs — CHANGES_REQUESTED** [review]

File: libs/api/domains/translations/src/lib/translations.service.ts (hunk 1 above)
Comment (magnearun) [reply libs/api/domains/translations/src/lib/translations.service.ts:36]: yes this fetches the field in all locales

File: libs/application/templates/reference-template/src/forms/messages.ts (hunk 3 above)
Comment (magnearun) [reply libs/application/templates/reference-template/src/forms/messages.ts:16]: No, I forgot I branched out of another branch, not master. That should be merged, will rebase

**Review by arnifreyrs — APPROVED** [review]

**Review by atligudl — APPROVED** [review]
Good upgrade!
I wonder if there's anything we can do about text flickering from the default message to the locale message on the initial fetch?
